test/packetimpact/tests/syn_with_same_ip_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syn_with_same_ip_test
+
+import (
+	"golang.org/x/sys/unix"
+	"gvisor.dev/gvisor/pkg/tcpip/header"
+	tb "gvisor.dev/gvisor/test/packetimpact/testbench"
+	"testing"
+	"time"
+)
+
+func TestSynWithSameLocalRemoteIP(t *testing.T) {
+	dut := tb.NewDUT(t)
+	listenFd, remotePort := dut.CreateListener(unix.SOCK_STREAM, unix.IPPROTO_TCP, 1)
+	defer dut.Close(listenFd)
+
+	// Set flag to have same LocalIP Address as RemoteIP.
+	tb.SetSameLocalRemoteIP = 1
+
+	conn := tb.NewTCPIPv4(t, dut, tb.TCP{DstPort: &remotePort}, tb.TCP{SrcPort: &remotePort})
+	defer conn.Close()
+
+	// Send SYN to DUT.
+	conn.Send(tb.TCP{Flags: tb.Uint8(header.TCPFlagSyn)})//, tb.IPv4{SrcAddr: })
```

Comment:
As described above, this will be `conn.SendFrame(...)` and you'll build the frame like how `conn.Send(...)` does but you'll override the source IP address to be the same as the dest IP address.

Reply:
[GES] Ok will take care as mentioned above.

Reply:
Another thing, could you split this into two PRs, one for each bug?  We should be keeping PRs small and these two bugs are unrelated.

Reply:
[GES] Ok. Will raise separate PR for each of the issue